### DISCUSSION TITLE
modify parameters with swmm-api and generate inp to simulate pyswmm

### DISCUSSION THREAD

**Discussion by alejorojas95** [pyswmm/pyswmm#544] [General]
implement the parameter modification exactly in the inp it is called Geom1, and in swmm-api it is called height (I call it diameter) it is linked to the XSECTION of the ducts or links or is there a way to correct this code or modify the parameter with pyswmm. thank you
```python
def cambiar_diametros(inp_file):
    inp = read_inp_file(inp_file)  # type: swmm_api.SwmmInput
    
    inp[SEC.XSECTIONS]["P1-P2"].height = 0.5
   
    inp.write_file('new_inputfile.inp')
```

estuve indagando y creo que con pyswmm hay solucion peor no logro modificar XSECTION
```python
def cambiar_diametros(inp_file):
    sim_conf = SimulationPreConfig()
    sim_conf.add_update_by_token("XSECTIONS", "P1-P2", 2, 0.8)
    #sim_conf.apply_changes()

    
    with Simulation(inp_file, sim_preconfig = sim_conf) as sim:
        cout= XSECTIONS(sim)
        print(cout)

        for step in sim:
            pass
```

I DON'T SEE MUCH DOCUMENTATION THAT SHOWS IT AND IT IS POSSIBLE TO GENERATE THE INP WITH THESE CHANGES I BETTER APPRECIATE YOUR RESPONSE

**Comment by bemcdonnell** [comment]
@alejorojas95, Great question!  When we released PySWMM-v2, we added the ability to modify the INP before it is opened with pyswmm.  Here is the [documentation](https://pyswmm.github.io/pyswmm/reference/api/pyswmm.simulation.SimulationPreConfig.html) using `SimulationPreConfig`.

In your case you will do the following:

`[XSECTIONS]` looks like this

```
[XSECTIONS]
;;Link           Shape        Geom1            Geom2      Geom3      Geom4      Barrels    Culvert
;;-------------- ------------ ---------------- ---------- ---------- ---------- ---------- ----------
```

Parameter index number for `GEOM1` is 2

```python
from pyswmm import Simulation, SimulationPreConfig

# Create Config Handle
sim_conf = SimulationPreConfig()

# Specifying the update parameters
# Parameter Order:
# Section, Object ID, Parameter Index, New Value, Obj Row Num (optional)
sim_conf.add_update_by_token("XSECTIONS", "conduit1", 2, 6)
sim_conf.add_update_by_token("XSECTIONS", "conduit2", 2, 6)

with Simulation(<path-to-inp>, sim_preconfig = sim_conf) as sim:
    for step in sim:
        pass
```

**Reply by bemcdonnell** [reply]
@alejorojas95, I just saw your edited comment.  This section of your code will fail: 

```python
        cout= XSECTIONS(sim)
        print(cout)
```
There is no XSECTIONS object as part of pyswmm.  Just to confirm the changes are being made, look at your INP inside of a text editor and you will see the changes.

**Reply by alejorojas95** [reply]
If this fails, thanks for the answer, look, I carried out a normal simulation.

If this fails, thanks for the answer, look, I carried out a normal simulation.

Then I do one to modify the parameter, but I don't see the changes reflected in the inp as text.

you create a new inp with _mod
I see
